File: solutions/julia/cheese-club/1/cheese-club.jl
function all_15(ratings)
    mapreduce(x-> x == 1 || x == 5, &, ratings)
end

function emphatics(customers)
    filter(x -> all_15(last(x)), customers) 
end

function tobinary(ratings)
    map(x -> x == 5 ? 1 : 0, ratings)
end

function tobinarymatrix(ratings)
    mapreduce(x -> transpose(tobinary(x)), vcat, ratings)
end

File: solutions/r/cheese-club/1/cheese-club.R
all_15 <- function(ratings) {
  all(ratings %in% c(1,5))
}

name_customers <- function(names, ratings) {
  Map(\(n, r) list(name = n, rating = r), names, ratings) |>
      unname()
}

emphatics <- function(names, ratings) {
  unname(Filter(\(x) all_15(x$rating), name_customers(names, ratings)))
}

to_binary <- function(ratings) as.integer(ratings == 5)

satisfactions <- function(ratings) {
  round(cumsum(ratings) / seq_along(ratings), 2)
}

